### PR TITLE
Added test for RPCs over a flaky network

### PR DESCRIPTION
**Pull request by rmstar**
Test the behavior of gRPC when the network is flaky
- network connection flaps
- traffic to gRPC server is blackholed
- Inject delays and packet loss on the client-server connection

### CODE REVIEW

File: test/cpp/end2end/BUILD [hunk 1]
```diff
@@ -531,6 +531,24 @@ grpc_cc_test(
     ],
 )
 
+grpc_cc_test(
```
Comment (jtattermusch) [inline test/cpp/end2end/BUILD]: this test needs to be set as "manual" for bazel, there's no chance of it working on foundry IMHO.

File: tools/run_tests/dockerize/build_and_run_docker.sh [hunk 2]
```diff
@@ -57,6 +57,7 @@ CONTAINER_NAME="build_and_run_docker_$(uuidgen)"
 docker run \
   "$@" \
   --cap-add SYS_PTRACE \
+  --cap-add NET_ADMIN \
```
Comment (jtattermusch) [inline tools/run_tests/dockerize/build_and_run_docker.sh]: I don't like the idea of adding a new capability for all X thousand tests when there's just one that needs this.

File: test/cpp/end2end/flaky_network_test.cc [hunk 3]
```diff
@@ -0,0 +1,413 @@
+/*
```
Comment (jtattermusch) [inline test/cpp/end2end/flaky_network_test.cc]: please get a C person to review this file.

File: tools/run_tests/dockerize/build_and_run_docker.sh (hunk 2 above)
Comment (rmstar) [reply tools/run_tests/dockerize/build_and_run_docker.sh]: What's a better way to do this that would only result in this one test running with NET_ADMIN capability?

File: test/cpp/end2end/flaky_network_test.cc (hunk 3 above)
Comment (rmstar) [reply test/cpp/end2end/flaky_network_test.cc]: Thanks, will do

File: test/cpp/end2end/BUILD (hunk 1 above)
Comment (rmstar) [reply test/cpp/end2end/BUILD]: Can you please point me to the configuration to set this to "manual"?

File: test/cpp/end2end/flaky_network_test.cc (hunk 3 above)
Comment (rmstar) [reply test/cpp/end2end/flaky_network_test.cc]: @muxi  can you please review the C++ test (this file)? The build stuff needs more work; I'm looking into that.

**Comment by rmstar** [comment]
@jtattermusch  PTAL. This test will only be built by manual bazel builds. The internal CI pipeline will run this test periodically.

File: test/cpp/end2end/flaky_network_test.cc (hunk 3 above)
Comment (muxi) [reply test/cpp/end2end/flaky_network_test.cc]: changes LGTM but I also suggest asking for another review from core, e.g. yang or nico? :)

**Comment by rmstar** [comment]
Known failure: ##17999

File: tools/internal_ci/linux/grpc_bazel_privileged_docker.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# Copyright 2017 gRPC authors.
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_bazel_privileged_docker.sh]: Update license to "Copyright 2019 The gRPC Authors" for new files.

Comment (rmstar) [reply tools/internal_ci/linux/grpc_bazel_privileged_docker.sh]: Thanks, fixed.

**Comment by rmstar** [comment]
@nicolasnoble is this ok to merge?

**Comment by rmstar** [comment]
I'll go ahead and merge this PR as it gives us test coverage for network flakiness.

**Pull request merged**